pkg/k8s/wrapper.go

```diff
@@ -0,0 +1,95 @@
+package k8s
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-app-mesh-controller-for-k8s/controllers/appmesh/custom"
+	v1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// K8sWrapper represents an interface with all the common operations on pod objects
```

Comment:
this feels more like a PodRepository instead of k8swrapper :D

Reply:
Tried to keep the naming convention in parity with VPC resource controller. I will update the name though